addons/web/static/src/core/l10n/translation.js

```diff
@@ -101,21 +108,38 @@ export async function loadLanguages(orm) {
 }
 
 /**
- * Same behavior as sprintf, but if any of the provided values is a markup,
- * escapes all non-markup content before performing the interpolation, then
- * wraps the result in a markup.
+ * Same behavior as sprintf, but doing two additional things:
+ * - If any of the provided values is an iterable, it will format its items
+ *   as a language-specific formatted string representing the elements of the
+ *   list.
+ * - If any of the provided values is a markup, it will escape all non-markup
+ *   content before performing the interpolation, then wraps the result in a
+ *   markup.
  *
  * @param {string} str The string with placeholders (%s) to insert values into.
  * @param  {...any} values Primitive values to insert in place of placeholders.
  * @returns {string|Markup}
  */
-function _safeSprintf(str, ...values) {
-    let hasMarkup;
+function _safeFormatAndSprintf(str, ...values) {
+    let hasMarkup = false;
+
+    const processTranslationArg = (key, value, obj) => {
+        if (isIterable(value)) {
```

Comment:
According to some of the issues on runbot I believe this excludes string but doesn't properly exclude markup.

Reply:
Ha, you're right. Good catch! I check now for `instanceof String`, since Markup is an inherited class of that.